scripts/linpack/cb_run_linpack.sh

```diff
@@ -22,47 +22,161 @@ source $(echo $0 | sed -e "s/\(.*\/\)*.*/\1.\//g")/cb_common.sh
+	HPL.out      output file name (if any)
+	6            device out (6=stdout,7=stderr,file)
+	1            # of problems sizes (N)
+	10240        Ns
+	1            # of NBs
+	240           NBs
+	0            PMAP process mapping (0=Row-,1=Column-major)
+	1            # of process grids (P x Q)
+	${P}            Ps
+	${Q}            Qs
+	16.0         threshold
+	1            # of panel fact
+	2            PFACTs (0=left, 1=Crout, 2=Right)
+	1            # of recursive stopping criterium
+	4            NBMINs (>= 1)
+	1            # of panels in recursion
+	2            NDIVs
+	1            # of recursive panel fact.
+	2            RFACTs (0=left, 1=Crout, 2=Right)
+	1            # of broadcast
+	2            BCASTs (0=1rg,1=1rM,2=2rg,3=2rM,4=Lng,5=LnM)
+	1            # of lookahead depth
+	1            DEPTHs (>=0)
+	1            SWAP (0=bin-exch,1=long,2=mix)
+	64           swapping threshold
+	0            L1 in (0=transposed,1=no-transposed) form
+	0            U  in (0=transposed,1=no-transposed) form
+	1            Equilibration (0=no,1=yes)
+	8            memory alignment in double (> 0)" > ${LINPACK_DAT}
+
+	CMDLINE="/opt/openmpi/bin/mpirun --allow-run-as-root -np $NR_CPUS --report-bindings ${LINPACK} > ${RUN_OUTPUT_FILE}"
+	execute_load_generator "$CMDLINE"
+	AVERAGE=$(cat ${RUN_OUTPUT_FILE} | grep -A 2 Gflops | awk '{print $7}' | tail -n1)
+        ~/cb_report_app_metrics.py \
+        throughput:$AVERAGE:gflops \
+        $(common_metrics)
+else
+	# Older Linpack
+	LINPACK=`get_my_ai_attribute_with_default linpack ~/compilers_and_libraries_2016.0.038/linux/mkl/benchmarks/linpack/xlinpack_xeon64`
+	eval LINPACK=${LINPACK}
+
+	sudo ls ${LINPACK} 2>&1 > /dev/null
+	if [[ $? -ne 0 ]]
+	then
+		LINPACK=$(sudo find ~ | grep xlinpack_xeon64)
+	fi
+	LOAD_FACTOR=`get_my_ai_attribute_with_default load_factor 5000`
+	LINPACK_DAT='~/linpack.dat'
+	eval LINPACK_DAT=${LINPACK_DAT}
+
+	PROBLEM_SIZES=$((${LOAD_LEVEL}*${LOAD_FACTOR}))
+	LEADING_DIMENSIONS=$((${LOAD_LEVEL}*${LOAD_FACTOR}))
+
+	LINPACK_IP=`get_ips_from_role linpack`
+
+	linux_distribution
+	export OMP_NUM_THREADS=$NR_CPUS
+	echo "Sample Intel(R) LINPACK data file (from lininput_xeon64)" > ${LINPACK_DAT}
+	echo "Intel(R) LINPACK data" >> ${LINPACK_DAT}
+	echo "1 # number of tests" >> ${LINPACK_DAT}
+	echo "$PROBLEM_SIZES # problem sizes" >> ${LINPACK_DAT}
+	echo "$LEADING_DIMENSIONS # leading dimensions" >> ${LINPACK_DAT}
+	echo "${LOAD_DURATION} # times to run a test " >> ${LINPACK_DAT}
+	echo "4 # alignment values (in KBytes)" >> ${LINPACK_DAT}
+
+	CMDLINE="${LINPACK} ${LINPACK_DAT}"
+
+	execute_load_generator "$CMDLINE" ${RUN_OUTPUT_FILE} ${LOAD_DURATION}
+	RESULTS=$(cat ${RUN_OUTPUT_FILE} | grep -A 1 Average | grep $PROBLEM_SIZES)
+	AVERAGE=$(echo $RESULTS | awk '{print $4}')
+	MAX=$(echo $RESULTS | awk '{print $5}')
     
+	~/cb_report_app_metrics.py \
+	throughput_max:$MAX:gflops \
+	throughput:$AVERAGE:gflops \
+	$(common_metrics)
+fi
```

Comment:
Could we just make the older version obsolete and remove the 2nd part of this?

Reply:
Yes thats actually easier. I was not sure for the case when someone would like to compare previous linpack run results? Maybe mark the new one default and have a way to run the older version if someone wants to use it? 

I am OK either way..

Reply:
Just updated it so 2.3 becomes the default version. Still keeping the older version so we have a way to transition between older and newer results. Let me know what you think.

Reply:
Looks great. I can see how radically different the two version in. Very nice patch. Thank you sir.